frontend/public/components/secrets/create-secret/PullSecretCredentialsForm.tsx

```diff
@@ -0,0 +1,151 @@
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { Base64 } from 'js-base64';
+import { Button } from '@patternfly/react-core';
+import { MinusCircleIcon } from '@patternfly/react-icons/dist/esm/icons/minus-circle-icon';
+import { PlusCircleIcon } from '@patternfly/react-icons/dist/esm/icons/plus-circle-icon';
+import { AUTHS_KEY, PullSecretCredentialEntry, PullSecretData } from '.';
+import { useEffect, useState } from 'react';
+
+export const PullSecretCredentialsForm: React.FC<PullSecretCredentialsFormProps> = ({
+  onChange,
+  stringData,
+}) => {
+  const { t } = useTranslation();
+  const isDockerconfigjson = _.isEmpty(stringData) || !!stringData[AUTHS_KEY];
+  const newImageSecretEntry = (): SecretEntry => ({
+    entry: {
+      address: '',
+      username: '',
+      password: '',
+      email: '',
+      auth: '',
+    },
+    uid: _.uniqueId(),
+  });
+  const imageSecretObjectToArray = (imageSecretObject): SecretEntry[] => {
+    const imageSecretArray = [];
+    if (_.isEmpty(imageSecretObject)) {
+      return _.concat(imageSecretArray, newImageSecretEntry());
+    }
+    _.each(imageSecretObject, (v, k) => {
+      // Decode and parse 'auth' in case 'username' and 'password' are not part of the secret.
+      const decodedAuth = Base64.decode(_.get(v, 'auth', ''));
+      const parsedAuth = _.isEmpty(decodedAuth) ? _.fill(Array(2), '') : _.split(decodedAuth, ':');
+      imageSecretArray.push({
+        entry: {
+          address: k,
+          username: _.get(v, 'username', parsedAuth[0]),
+          password: _.get(v, 'password', parsedAuth[1]),
+          email: _.get(v, 'email', ''),
+          auth: _.get(v, 'auth', ''),
+        },
+        uid: _.get(v, 'uid', _.uniqueId()),
+      });
+    });
+    return imageSecretArray;
+  };
+  const imageSecretArrayToObject = (imageSecretArray: SecretEntry[]) => {
+    const imageSecretsObject = {};
+    _.each(imageSecretArray, (value) => {
```

Comment:
Use Array.prototype.reduce instead of _.each
```suggestion
    const imageSecretsObject = {};
    _.each(imageSecretArray, (value) => {
```